### PR TITLE
IMI-2282 - feat: adding field to support storing final results publicly

### PR DESCRIPTION
**Pull request by hugoseabra-imachines**
Adding field `public_results` in Manifest model to support this option to store final results when not encrypted.

IMI-2282

### CODE REVIEW

File: basemodels/manifest/manifest.py [hunk 1]
```diff
@@ -176,6 +176,7 @@ class Manifest(Model):
     job_total_tasks = IntType(required=True)
     network = StringType(required=False)
     only_sign_results = BooleanType(default=False,)
+    store_pub_final_results = BooleanType(default=False,)
```
Comment (DicksonChi) [inline basemodels/manifest/manifest.py:179]: The default for BooleanType is already False no need to add that

Comment (hugoseabra-imachines) [reply basemodels/manifest/manifest.py:179]: I also thought so. But checking the `BooleanType` class, there is nothing to determine a default value. Check [this](https://github.com/schematics/schematics/blob/master/schematics/types/base.py#L598). So, if nothing is passed, the default value will be `None`, which is not acceptable.

Comment (DicksonChi) [reply basemodels/manifest/manifest.py:179]: yes you are right!

**Review by DicksonChi — APPROVED** [review]
LGTM

**Review by e271828- — CHANGES_REQUESTED** [review]
Please rename to `public_results`

**Comment by spirit-q2** [comment]
I still don't understand why exactly we want this as a new DB field and not as the manifest key... 🤔
can you remind me of the reason?

cc @igervasiychuk

File: basemodels/manifest/manifest.py [hunk 2]
```diff
@@ -176,6 +176,7 @@ class Manifest(Model):
     job_total_tasks = IntType(required=True)
     network = StringType(required=False)
     only_sign_results = BooleanType(default=False,)
+    public_results = BooleanType(default=False,)
```
Comment (spirit-q2) [inline basemodels/manifest/manifest.py:179]: I still don't understand why exactly we want this as a new DB field and not as the manifest key... 🤔
can you remind me of the reason?

cc @igervasiychuk

Comment (igervasiychuk) [reply basemodels/manifest/manifest.py:179]: the reasoning was that it **is** a manifest key, but we also copy the most important fields into columns. And that makes access to them easier downstream (you can look at other relevant PRs).

**Review by DicksonChi — APPROVED** [review]
LGTM

**Review by e271828- — APPROVED** [review]

**Pull request merged**